test/test_indexing.py

```diff
@@ -779,6 +779,16 @@ def test_index_put_accumulate_duplicate_indices(self, device):
 
             self.assertEqual(output, input_list)
 
+    @onlyCUDA
+    def test_index_put_deterministic(self, device):
+        a = torch.arange(10, dtype=torch.float, device=device)
+        a[a>2.] = 1
+        torch.use_deterministic_algorithms(True) 
+        b = torch.arange(10, dtype=torch.float, device=device)
+        b[b>2.] = 1
+        torch.use_deterministic_algorithms(False)
+        self.assertEqual(a, b)
+
```

Comment:
Would it make sense to check, if deterministic algorithms are enabled at the beginning of the test via `torch.are_deterministic_algorithms_enabled()` and reset it to this value before exiting this test?
This would make sure to keep the "global" setup in case it was changed before (default should be `False` in any case).

Reply:
Definitely! Typically deterministic algos are tested within a context manager to preserve default setting.